formats/roam/roam.ts

```diff
@@ -0,0 +1,327 @@
+
+	blockText = blockText.replaceAll(/{{.*?\bvideo\b.*?(\bhttp.*?\byoutu.*?)}}/g, "![]($1)"); // youtube embeds
+	blockText = blockText.replaceAll(/(https?:\/\/twitter\.com\/(?:#!\/)?\w+\/status\/\d+(?:\?[\w=&-]+)?)/g, "![]($1)"); // twitter embeds
+	blockText = blockText.replaceAll(/{{.*?roam\/render[^}]*}}/g, ""); // {{roam/render}} components
+	blockText = blockText.replaceAll(/\_\_(.+?)\_\_/g, "*$1*"); // __ __ itallic
+	blockText = blockText.replaceAll(/\^\^(.+?)\^\^/g, "==$1=="); // ^^ ^^ highlight
+    
+	// block and page embeds
+	blockText = blockText.replaceAll(/{{\[{0,2}embed.*?(\(\(.*?\)\)).*?}}/g, "$1")
+	blockText = blockText.replaceAll(/{{\[{0,2}embed.*?(\[\[.*?\]\]).*?}}/g, "$1")
+	// download files uploaded to Roam
+	if (downloadAttachments) {
+		if (blockText.includes('firebasestorage')) {
+			// console.log(app)
+			blockText =  await downloadFirebaseFile(blockText, attachmentsFolder)
+		}
+	}
+    // blockText = blockText.replaceAll("{{[[table]]}}", ""); 
+	// blockText = blockText.replaceAll("{{[[kanban]]}}", "");
+	// blockText = blockText.replaceAll("{{mermaid}}", "");
+	// blockText = blockText.replaceAll("{{[[mermaid]]}}", "");
+    // blockText = blockText.replaceAll("{{diagram}}", "");
+	// blockText = blockText.replaceAll("{{[[diagram]]}}", "");
+
+	//   blockText = blockText.replaceAll(/\!\[(.+?)\]\((.+?)\)/g, "$1 $2"); //images with description
+	//   blockText = blockText.replaceAll(/\!\[\]\((.+?)\)/g, "$1"); //imags with no description
+	//   blockText = blockText.replaceAll(/\[(.+?)\]\((.+?)\)/g, "$1: $2"); //alias with description
+	//   blockText = blockText.replaceAll(/\[\]\((.+?)\)/g, "$1"); //alias with no description
+	//   blockText = blockText.replaceAll(/\[(.+?)\](?!\()(.+?)\)/g, "$1"); //alias with embeded block (Odd side effect of parser)
+
+	return blockText;
+  };
+
+async function jsonToMarkdown(graphFolder:string, attachmentsFolder:string, downloadAttachments:boolean,json: JsonObject, indent: string = '', isChild: boolean = false ): Promise<string> {
+    let markdown = '';
+
+    if (json.string) {
+        let prefix = '';
+        if (json.heading) {
+            prefix = '#'.repeat(json.heading) + ' ';
+        }
+        markdown += `${isChild ? indent + '* ' : indent}${prefix}${(await roamMarkupScrubber(graphFolder, attachmentsFolder, json.string, downloadAttachments))}\n`;
+    }
+
+    if (json.children) {
+        for (const child of json.children) {
+            markdown += await jsonToMarkdown(graphFolder, attachmentsFolder, downloadAttachments, child, indent + '  ', true);
+        }
+    }
+
+    return markdown;
+}
+
+export async function importRoamJson (importer:RoamJSONImporter, progress: ProgressReporter, files:PickedFile[], folder:TFolder, downloadAttachments:boolean = true) {
+    for (let file of files) {
+        const graphName = sanitizeFileName(file.basename);
+        const graphFolder = path.join(folder.path, graphName);
```

Comment:
`path.join` doesn't work on mobile, plus for internal use we can simply put it between `/`.